### DISCUSSION TITLE
Re-adding files in directory

### DISCUSSION THREAD

**Discussion by axhamre** [twpayne/chezmoi#2742] [Q&A] [answered]
This is the description of the re-add command:

> Re-add all modified files in the target state. chezmoi will not overwrite templates, and **all entries that are not files are ignored**.

Does the last part mean that files in directories will not work for `re-add`, it only works for files in root? If that's correct assumption, is there any workaround for this - i.e. a command that adds/re-adds any target file that have been modified?

**Comment by twpayne** [comment, marked as answer]
`chezmoi re-add` does recurse into subdirectories, re-adding any modified files that are not templates.

You can run `chezmoi re-add --interactive` to be prompted for each file. You can also run `chezmoi add --dry-run --verbose` to see the changes that chezmoi would make to the source state.
